### DISCUSSION TITLE
Is there any way to filter symbols in Symbol Picker for one type of objects?

### DISCUSSION THREAD

**Discussion by nxy7** [helix-editor/helix#8948] [Q&A] [answered]
I think using workspace symbol picker is good way to understand new codebases, something that can be really useful would be filtering symbols for selected types, f.e. only golang interfaces or rust traits. I've looked through documentation, but it seems like there's no such feature right now (unless I'm mistaken), maybe it would be good idea to add something like that?
I imagine even character flag (s - for objects/structs, f - for functions, t- for traits/interfaces) would be good start. Using character flags has this issue that some names do not map between languages, but using rust names would make sense (especially since helix is written in rust).

**Comment by the-mikedavis** [comment, marked as answer]
Not currently. There's some work in progress to refactor the picker to be a table where the symbol kind becomes one of the columns and you can filter on each column individually: https://github.com/helix-editor/helix/pull/7265
